### PR TITLE
Fix logs disappearing after filtering when console buffer overflowed at least once

### PR DESCRIPTION
**Pull request by LeoMash**

### CODE REVIEW

File: src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/toolWindow/UnrealToolWindowFactory.kt [hunk 1]
```diff
@@ -232,95 +201,58 @@ class UnrealToolWindowFactory(val project: Project)
...
+        for (rangeWithIndex in allRanges.withIndex()) {
+            val match = line.substring(rangeWithIndex.value.first, rangeWithIndex.value.last)
+            if (rangeWithIndex.value in unrealLogEvent.bpPathRanges) {
+                val hyperLinkInfo = BlueprintClassHyperLinkInfo(model.openBlueprint, BlueprintReference(FString(match)))
+                consoleView.printHyperlink(match, hyperLinkInfo)
+            } else {
+                val (`class`, method) = match.split(MethodReference.separator)
+                val methodReference = MethodReference(UClass(FString(`class`)), FString(method))
+
+                val classHyperLinkInfo = UnrealClassHyperLinkInfo(model, methodReference, UClass(FString(`class`)))
```
Comment (yvvan) [inline src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/toolWindow/UnrealToolWindowFactory.kt:222]: Won't that create a hyperlink which leads nowhere if it's not a valid class::method?

**Review by yvvan — COMMENTED** [review]
I really like your solution! Have only once concern but probably I don't understand properly how hyperlink's 'navigate' works.

File: src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/toolWindow/UnrealToolWindowFactory.kt (hunk 1 above)
Comment (LeoMash) [reply src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/toolWindow/UnrealToolWindowFactory.kt:222]: Yes, now we will create false hyperlinks sometimes, but it's the only solution I came with to not rely on offset calculation in console text buffer that is very unstable.
I think it's good enough compromise for much simpler code and stable behavior.

Comment (yvvan) [reply src/rider/main/kotlin/com/jetbrains/rider/plugins/unreal/toolWindow/UnrealToolWindowFactory.kt:222]: I see. I think it's an acceptable solution for now.

**Review by DecoyRS — APPROVED** [review]
LGTM

**Pull request merged**